File: src/main/java/bot/util/database/DatabaseConnection.java
package bot.util.database;

import bot.util.Log;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class DatabaseConnection {

    private final Log log = new Log();

    private final String HOST;
    private final String PORT;
    private final String USER;
    private final String PASS;
    private final String NAME;

    public DatabaseConnection() {
        this.HOST = System.getenv("HOST");
        this.PORT = System.getenv("PORT");
        this.USER = System.getenv("USER");
        this.PASS = System.getenv("PASS");
        this.NAME = System.getenv("NAME");
    }

    public Connection getConnection(){
        try {
            Class.forName("org.postgresql.Driver");
            String URL = "jdbc:postgresql://" + HOST + ":" + PORT + "/" + NAME;
            log.databaseConnect(URL, PASS, USER);
            return DriverManager.getConnection(URL, USER, PASS);
        } catch (ClassNotFoundException | SQLException e) {
            log.userException(e);
            return null;
        }
    }
}

File: src/main/java/bot/util/database/DatabaseConnection.kt
package bot.util.database

import bot.util.Log
import java.sql.Connection
import java.sql.DriverManager
import java.sql.SQLException
import kotlin.system.exitProcess

class DatabaseConnection {
    private val log = Log()

    private var HOST: String? = System.getenv("HOST")
    private var USER: String? = System.getenv("USER")
    private var PASS: String? = System.getenv("PASS")
    private var NAME: String? = System.getenv("NAME")

    fun getConnection(): Connection {
        val URL = "jdbc:postgresql://$HOST:5432/$NAME"
        return try {
            Class.forName("org.postgresql.Driver")
            log.databaseConnect(URL, PASS, USER)
            DriverManager.getConnection(URL, USER, PASS)
        } catch (e: ClassNotFoundException) {
            log.userException(e)
            log.databaseConnectionError(URL, USER, PASS)
            exitProcess(1)
        } catch (e: SQLException) {
            log.userException(e)
            log.databaseConnectionError(URL, USER, PASS)
            exitProcess(1)
        }
    }
}